tests/nativeapp/test_sf_sql_facade.py

```diff
@@ -0,0 +1,88 @@
+# Copyright (c) 2024 Snowflake Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from unittest import mock
+
+from snowflake.cli._plugins.nativeapp.sf_facade import get_snowflake_facade
+from snowflake.cli.api.entities.common import get_sql_executor
+
+from tests.nativeapp.utils import (
+    SQL_EXECUTOR_EXECUTE,
+    SQL_EXECUTOR_EXECUTE_QUERIES,
+    mock_execute_helper,
+)
+
+# TODO: Add more test coverage + define contract
+# TODO: Add test for non-identifier names
+sql_facade = get_snowflake_facade(get_sql_executor())
+
+
+@mock.patch(SQL_EXECUTOR_EXECUTE)
+@mock.patch(SQL_EXECUTOR_EXECUTE_QUERIES)
+def test_execute_with_role_wh_db(mock_execute_queries, mock_execute_query, mock_cursor):
+    # Arrange
+    mock_script = "-- my comment\nselect 1;\nselect 2;"
+    mock_script_name = "test-user-sql-script.sql"
+    role = "mock_role"
+    wh = "mock_wh"
+    database = "mock_db"
+    side_effects, expected = mock_execute_helper(
```

Comment:
if you have a better pattern, use it instead. This is pretty much NADE legacy at this point. We could also create a new abstraction / helper for SQL testing in this file, since it will be used here almost exclusively.

Reply:
I think using this helper is OK for when we actually want to test the sql. 
And its usage in other tests will be reduced and removed as I refactor more SQL calls into this class.
Of course there are small improvements I can do to it like switching the order of expected call and side_effect but not sure how much we'll gain from that.